pkgs/applications/networking/bee/bee-clef.nix

```
@@ -0,0 +1,71 @@
+{ version ? "release", stdenv, lib, substituteAll
+, fetchFromGitHub, go-ethereum }:
+
+let
+
+  versionSpec = {
+    release = rec {
+      name = "bee-clef";
+      version = "0.4.4";
+      rev = "refs/tags/v${version}";
+      sha256 = "18fab0nxxsglrdykhb02m6vd7pgd6z4pqgfp5nmpxjcz7s8rfnml";
+    };
+  }.${version};
+
+in
+
+stdenv.mkDerivation {
+  pname = versionSpec.name;
+  version = versionSpec.version;
+
+  src = fetchFromGitHub {
+    owner = "ethersphere";
+    repo = "bee-clef";
+    rev = versionSpec.rev;
+    sha256 = versionSpec.sha256;
+  };
+
+  buildInputs = [ go-ethereum ];
+
+  patches = [
+    (substituteAll {
+      src = ./bee-clef.patch;
+      clef = "${go-ethereum}/bin/clef";
+      inherit outputDir;
+
+      # this fails, but i also need to somehow inject paths that are referring to files in $out 
+      #??? inherit out;
+    })
+  ];
```

I think you can replace the patch altogether with `substituteInPlace`, I think.

Also I'd recommend installing `rules.js` and `4byte.json` to `$out/share/clef` or similar, since installing stuff into top level `$out` is not nice and can cause unwanted conflicts in the worst cases.

```suggestion
  postPatch = ''
    substituteInPlace packaging/clef-service \
      --replace "clef --stdio-ui" "$out/bin/clef --stdio-ui" \
      --replace "/etc/bee-clef" "$out"
  '';
```

---

the reason i was advised to use a patch file is to fail loudly when the upstream changes.

also note that the upstream files do not contain clear markers for replacements, so a `substituteInPlace` would be much more fragile. the patch file is there basically to add those markers in a failsafe way.

initially i used `sed`, which is practically a `substituteAll`, when i was advised to add a patch file.

the rest i'll do now, thank you!

---

Either way works, but I think achieving the same with a patch is probably more cumbersome and using `substituteInPlace` for fixing paths is pretty common practice.

What you prefer.